tests/functional/pv/pvc_clone/test_clone_deletion_after_max_cluster_space_utilization.py

```diff
@@ -0,0 +1,197 @@
+
+    def verify_alerts_via_prometheus(self, expected_alerts, threading_lock):
+        """
+        Verify Alerts on prometheus
+
+        Args:
+            expected_alerts (list): list of alert names
+            threading_lock (threading.Rlock): Lock object to prevent simultaneous calls to 'oc'
+
+        Returns:
+            bool: True if expected_alerts exist, False otherwise
+
+        """
+        prometheus = PrometheusAPI(threading_lock=threading_lock)
+        logger.info("Logging of all prometheus alerts started")
+        alerts_response = prometheus.get(
+            "alerts", payload={"silenced": False, "inhibited": False}
+        )
+        actual_alerts = list()
+        for alert in alerts_response.json().get("data").get("alerts"):
+            actual_alerts.append(alert.get("labels").get("alertname"))
+            print("Actual Alerts:", actual_alerts)
+        for expected_alert in expected_alerts:
+            if expected_alert not in actual_alerts:
+                logger.error(
+                    f"{expected_alert} alert does not exist in alerts list."
+                    f"The actaul alerts: {actual_alerts}"
+                )
+                return False
+        return True
+
+    @skipif_external_mode
+    @magenta_squad
+    def test_clone_deletion_after_max_cluster_space_utilization(
+        self, interface_type, pvc_clone_factory, threading_lock
+    ):
+        """
+        Steps:
+            1. Have a cluster with OCP + ODF internal mode
+            2. Create cephfs + RBD pvcs with data
+            3. Create clones of pvcs till cluster full ratio.
+            4. Make the cluster out of full by increasing the full ratio.
+            5. After the cluster is out of full state and IOs started , Try to delete clones.
+            6. Clone deletion should be successful and should not give error messages.
+        """
+        # Creating the clones one by one and wait until they bound
+        self.timeout = 1800
+        logger.info(
+            f"Start creating {self.num_of_clones} clones on {interface_type} PVC of size {self.pvc_size} GB."
+        )
+        clones_list = []
+        for i in range(self.num_of_clones):
+            index = i + 1
+            logger.info(f"Start creation of clone number {index}.")
+
+            cloned_pvc_obj = pvc_clone_factory(
+                self.pvc_obj,
+                storageclass=self.pvc_obj.backed_sc,
+            )
+
+            helpers.wait_for_resource_state(
+                cloned_pvc_obj, constants.STATUS_BOUND, timeout=1800
+            )
+            cloned_pvc_obj.reload()
+            clones_list.append(cloned_pvc_obj)
+            logger.info(
+                f"Clone with name {cloned_pvc_obj.name} for {self.pvc_size} pvc {self.pvc_obj.name} was created."
+            )
+
+        logger.info("Verify 'CephClusterErrorState' Alerts are seen ")
+
+        expected_alerts = "CephClusterErrorState"
```

Comment:
CephClusterErrorState alert is seen when the Storage cluster is in error state for more than 10m.
Before this, should we also check for CephClusterCriticallyFull alerts?

Reply:
No.. As per BZ i customer tried to delete clones after this stage.

Reply:
If the cluster is reaching the nearfull ratio, it is expected to get the  CephClusterCriticallyFull alerts